File: CS/CustomCachedDocumentSourceSerialization/CustomWebDocumentViewerOperationLogger.cs
using System;
using System.Collections.Generic;
using DevExpress.XtraPrinting;
using DevExpress.XtraPrinting.Caching;
using DevExpress.XtraReports.Web.WebDocumentViewer;

namespace CustomCachedDocumentSourceSerialization {
    public class CustomWebDocumentViewerOperationLogger: WebDocumentViewerOperationLogger {
        public override void CachedDocumentSourceSerializing(string documentId, CachedDocumentSource cachedDocumentSource, GeneratedDocumentDetails documentDetails, DocumentStorage documentStorage, PrintingSystemBase printingSystemSource) {
            var serviceProvider = printingSystemSource as IServiceProvider;
            var customPageDataService = serviceProvider.GetService(typeof(CustomPageDataService)) as CustomPageDataService;
            if (customPageDataService != null) {
                documentDetails.CustomData = new Dictionary<string, object> { [CustomPageDataService.Key] = customPageDataService.PageAdditionalData };
                cachedDocumentSource.PrintingSystem.XlSheetCreated += customPageDataService.PrintingSystem_XlSheetCreated;
            }
        }

        public override void CachedDocumentSourceDeserialized(string documentId, CachedDocumentSource cachedDocumentSource, GeneratedDocumentDetails documentDetails, DocumentStorage documentStorage) {
            object customData;
            if(documentDetails.CustomData != null && documentDetails.CustomData.TryGetValue(CustomPageDataService.Key, out customData)){
                Dictionary<int, List<int>> customDataServiceDictionary = customData as Dictionary<int, List<int>>;
                if (customDataServiceDictionary == null)
                    return;
                var customPageDataService = new CustomPageDataService(customDataServiceDictionary);
                cachedDocumentSource.PrintingSystem.XlSheetCreated += customPageDataService.PrintingSystem_XlSheetCreated;
            }
        }
    }
}

File: VB/CustomCachedDocumentSourceSerialization/CustomWebDocumentViewerOperationLogger.vb
Imports System
Imports System.Collections.Generic
Imports DevExpress.XtraPrinting
Imports DevExpress.XtraPrinting.Caching
Imports DevExpress.XtraReports.Web.WebDocumentViewer

Namespace CustomCachedDocumentSourceSerialization
	Public Class CustomWebDocumentViewerOperationLogger
		Inherits WebDocumentViewerOperationLogger

		Public Overrides Sub CachedDocumentSourceSerializing(ByVal documentId As String, ByVal cachedDocumentSource As CachedDocumentSource, ByVal documentDetails As GeneratedDocumentDetails, ByVal documentStorage As DocumentStorage, ByVal printingSystemSource As PrintingSystemBase)
			Dim serviceProvider = TryCast(printingSystemSource, IServiceProvider)
			Dim customPageDataService = TryCast(serviceProvider.GetService(GetType(CustomPageDataService)), CustomPageDataService)
			If customPageDataService IsNot Nothing Then
				documentDetails.CustomData = New Dictionary(Of String, Object) From {
					{CustomCachedDocumentSourceSerialization.CustomPageDataService.Key, customPageDataService.PageAdditionalData}
				}
				AddHandler cachedDocumentSource.PrintingSystem.XlSheetCreated, AddressOf customPageDataService.PrintingSystem_XlSheetCreated
			End If
		End Sub

		Public Overrides Sub CachedDocumentSourceDeserialized(ByVal documentId As String, ByVal cachedDocumentSource As CachedDocumentSource, ByVal documentDetails As GeneratedDocumentDetails, ByVal documentStorage As DocumentStorage)
			Dim customData As Object = Nothing
			If documentDetails.CustomData IsNot Nothing AndAlso documentDetails.CustomData.TryGetValue(CustomPageDataService.Key, customData) Then
				Dim customDataServiceDictionary As Dictionary(Of Integer, List(Of Integer)) = TryCast(customData, Dictionary(Of Integer, List(Of Integer)))
				If customDataServiceDictionary Is Nothing Then
					Return
				End If
				Dim customPageDataService = New CustomPageDataService(customDataServiceDictionary)
				AddHandler cachedDocumentSource.PrintingSystem.XlSheetCreated, AddressOf customPageDataService.PrintingSystem_XlSheetCreated
			End If
		End Sub
	End Class
End Namespace